dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/transport/ChannelHandlerDispatcher.java

```diff
@@ -41,12 +43,15 @@ public ChannelHandlerDispatcher() {
     }
 
     public ChannelHandlerDispatcher(ChannelHandler... handlers) {
+        // if varargs is used, the type of handlers is ChannelHandler[] and it is not null
+        // so we should filter the null object
         this(handlers == null ? null : Arrays.asList(handlers));
     }
 
     public ChannelHandlerDispatcher(Collection<ChannelHandler> handlers) {
         if (CollectionUtils.isNotEmpty(handlers)) {
-            this.channelHandlers.addAll(handlers);
+            // filter null object
+            this.channelHandlers.addAll(handlers.stream().filter(Objects::nonNull).collect(Collectors.toSet()));
```

Comment:
```suggestion
            handlers.stream().filter(Objects::nonNull).foreach(this.channelHandlers::add);
```

Reply:
the channelHandlers is a CopyOnWriteArraySet, when we edit the value , CopyOnWriteArraySet will create a new list, so I think that we should use addAll